frontend/src/features/comparisons/statsSlice.ts

```diff
@@ -0,0 +1,30 @@
+import { RootState } from '../../app/store';
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+import { PollStats } from 'src/utils/types';
+import { DEFAULT_POLL_STATS } from 'src/utils/api/stats';
+
+export const initialState: PollStats = DEFAULT_POLL_STATS;
+
+export const statsSlice = createSlice({
```

Comment:
This slice will contain only the stats of one poll at a time for now, but the name `statsSlice` could be confusing. 

In the back end API, the statistics are represented with a JSON containing several polls in addition to the number of users. Ex:

```json
{
  "active_users": {
    "total": 0,
    "joined_last_month": 0,
    "joined_last_30_days": 0
  },
  "polls": [
    {
      "name": "string",
      "compared_entities": {
        "total": 0,
        "added_last_month": 0,
        "added_last_30_days": 0
      },
      "comparisons": {
        "total": 0,
        "added_last_month": 0,
        "added_last_30_days": 0,
        "added_current_week": 0
      }
    }
  ]
}
```

We have two options here:
- **(a)** saving the complete JSON object returned by the API in the store, which include all polls stats
- **(b)** saving only one poll at a time, and renaming the slice `pollStatsSlice` (singular)

The options **(a)** has my preference for few reasons:
- when we query the back end, it already answers with the statistics of all polls without any filter (so it doesn't cost anything to save everything)
- it opens the possibility to display the stats of different polls on the same page
- it makes the statsSlice, a real statsSlice, in sync with the stats API (and not a subpart of a response)
- if a developer wonder if some stats are available in the store, the answer will always be yes : )

I don't have a strong preference, it will be fine to have only one poll for now in this PR. You can choose @abhiram11  : )

Reply:
Hi @GresilleSiffle , I'm going forward with the **(a)** approach, should we rename the entire `PollStats` to something like `Stats` or `AllStats` ? I have:

1. Separated the entire component into an `<CollectiveGoalWeeklyProgress>` and placed it in `src/features/goals`.
2. The "PollStats" now contains the entire data received from "/stats/" API in a spreaded format.
3. Working on removing states in `HomeVideos, etc.` that were the alternative from redux slice to store the `/stats/` api data. This would also affect files such as `UsageStatsSection` as well.
4. The UI has also been minimalized as per your requirement, I would prefer with going with the Tooltip approach afterwards since it encourages client/user's curiosity to hover on their "Green coloured +60" contribution : )

Reply:
If you have chosen the approach **(a)** then we can go with:
- `statisticsSlice` for the Redux slice
-  `Statistics` for the type of the initial state (it's a type already available, you can import it with `import { Statistics } from 'src/services/openapi';`)

> Separated the entire component into an <CollectiveGoalWeeklyProgress> and placed it in src/features/goals.

Perfect.

> The "PollStats" now contains the entire data received from "/stats/" API in a spreaded format. Otherwise, keeping the "active_users", "compared_entities", etc. Type Definitions in the redux would introduce complexity, but let me know if you would want it that way.

Good. It matches the behaviour of the slice `userSettingsSlice` :+1: 

> Working on removing states in HomeVideos, etc. that were the alternative from redux slice to store the /stats/ api data. This would also affect files such as UsageStatsSection as well.

Exactly. You are not forced to do it in the same pull request, but it will be useful. The components previously using the state will need to use Redux selectors. You can find an example here:
- usage: https://github.com/tournesol-app/tournesol/blob/0dcf9fa5dd1674a35b9f757e9917c6d828180479/frontend/src/features/settings/preferences/VideosPollUserSettingsForm.tsx#L36
- definition: https://github.com/tournesol-app/tournesol/blob/0dcf9fa5dd1674a35b9f757e9917c6d828180479/frontend/src/features/settings/userSettingsSlice.ts#L33

> The UI has also been minimalized as per your requirement, I would prefer with going with the Tooltip approach afterwards since it encourages client/user's curiosity to hover on their "Green coloured +60" contribution : )

Great, let's see how the first implementation looks like.

Do not hesitate to ping me again when you are ready for a review ; )

Reply:
Sure, I can move forward with these ideas since most of it is already done. I'm currently dealing with memory leak issues in the useEffects where I'm calling `/stats/` api... along with some changes. I'll push a commit in an hour for your reference!

Reply:
We can later use cache and stale times effectively to call the `stats` API from components like `UsageStatsSection`, `PublicDownloadsSection`, and `CollectiveWeeklyProgress`, etc.